sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,252 @@
+                )
+        return self._client
+
+    @log_exceptions_and_usage(online_store="hazelcast")
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        fv_map = client.get_map(_table_id(config.project, table))
+
+        for entity_key, values, event_ts, created_ts in data:
+            entity_key_bin = serialize_entity_key(
+                entity_key,
+                entity_key_serialization_version=config.entity_key_serialization_version,
+            ).hex()
+            event_ts_utc = _to_utc_timestamp(event_ts)
+            created_ts_utc = 0.0
+            if created_ts is not None:
+                created_ts_utc = _to_utc_timestamp(created_ts)
+            for feature_name, value in values.items():
+                feature_value = value.SerializeToString().hex()
+                __key = str(entity_key_bin) + feature_name
+                fv_map.put(
+                    __key,
+                    HazelcastJsonValue(
+                        {
+                            "entity_key": entity_key_bin,
+                            "feature_name": feature_name,
+                            "feature_value": feature_value,
+                            "event_ts": event_ts_utc,
+                            "created_ts": created_ts_utc,
+                        }
+                    ),
+                    config.online_store.key_ttl_seconds,
+                )
+                if progress:
+                    progress(1)
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        entries: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+        fv_map = client.get_map(_table_id(config.project, table))
+
+        hz_keys = []
+        entity_keys_str = {}
+        for entity_key in entity_keys:
+            feature_keys = []
+            entity_key_str = str(
+                serialize_entity_key(
+                    entity_key,
+                    entity_key_serialization_version=config.entity_key_serialization_version,
+                ).hex()
+            )
+            if requested_features:
+                for feature in requested_features:
```

Comment:
You can use a list comprehension here, which is usually faster and more compact. That also allows you to get rid of `feature_keys = []` line:
```
feature_keys = [entity_key_str + feature for feature in requested_features]
```
The same can be used for the `else` branch.

Reply:
Makes sense, refactored at https://github.com/mehmettokgoz/feast/pull/1/commits/9f8174c72faf2bbdbd6f7fe4b414d63f20338600#diff-47764bf85c9631f63e542e314d1d29579a85e570b2813040caaade67743322d7R215